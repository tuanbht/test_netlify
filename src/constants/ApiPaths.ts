export const API_URL = process.env.REACT_APP_BASE_API;

export const ORDER_DETAILS_PATH = '/orders/:id/payment/';
export const MARK_AS_PAID_PATH = '/orders/:id/mask_as_paid/';
export const CANCEL_ORDER_PATH = '/orders/:id/payment/cancel/';
