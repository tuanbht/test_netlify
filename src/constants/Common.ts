export const VIEW_CRYPTO_TRANSACTION = process.env.REACT_APP_VIEW_CRYPTO_TRANSACTION;
