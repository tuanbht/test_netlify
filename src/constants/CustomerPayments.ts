import React from 'react';
import { ReactComponent as IconBTC } from '../assets/images/icons/icon-btc.svg';
import { ReactComponent as IconETH } from '../assets/images/icons/icon-eth.svg';
import { ReactComponent as IconUSDT } from '../assets/images/icons/icon-usdt.svg';

export const STEPS = {
  CRYPTO: 0,
  WALLET: 1,
  PROCESSING: 2,
  COMPLETE: 3,
};

export interface CryptoCurrencies {
  shortName: string;
  fullName: string;
  logo: React.FC<React.SVGProps<SVGSVGElement>>;
}

export class CryptoCurrencies implements CryptoCurrencies {
  constructor() {
    this.fullName = '';
    this.shortName = '';
  }
}

export const CRYPTO_CURRENCIES = {
  BITCOIN: {
    shortName: 'BTC',
    fullName: 'Bitcoin',
    logo: IconBTC,
  } as CryptoCurrencies,
  ETHEREUM: {
    shortName: 'ETH',
    fullName: 'Ethereum',
    logo: IconETH,
  } as CryptoCurrencies,
  USDT: {
    shortName: 'USDT',
    fullName: 'USD Tether',
    logo: IconUSDT,
  } as CryptoCurrencies,
};
