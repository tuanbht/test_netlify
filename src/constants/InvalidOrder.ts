export const ORDER_CANCELLED_CONTENT = {
  title: 'Order cancelled',
  subtitle: 'This order has been cancelled and is no longer available. To place a new order, contact your Merchant.',
};

export const ORDER_EXPIRED_CONTENT = {
  title: 'Order expired',
  subtitle: 'This order expired and is no longer available. To place a new order, contact your Merchant.',
};
