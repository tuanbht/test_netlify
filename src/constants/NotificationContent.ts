export const ORDER_CANCELLED_CONTENT = {
  title: 'Order cancelled',
  subtitle: 'This order has been cancelled and is no longer available. To place a new order, contact your Merchant.',
};

export const ORDER_EXPIRED_CONTENT = {
  title: 'Order expired',
  subtitle: 'This order expired and is no longer available. To place a new order, contact your Merchant.',
};
export const ROOT_PAGE_CONTENT = {
  title: 'Welcome to Paloma!',
  subtitle:
    "To pay for your order with Crypto, follow the link sent via text message or ask your merchant to initiate the payment process if you haven't yet received a text.",
};
