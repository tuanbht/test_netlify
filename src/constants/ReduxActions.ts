const SUCCESS_SUFFIX = '_SUCCESS';
const FAILURE_SUFFIX = '_FAIL';

export const GET_ORDER_DETAILS = 'GET_ORDER_DETAILS';
export const MARK_AS_PAID_ORDER = 'MARK_AS_PAID_ORDER';
export const CANCEL_ORDER = 'CANCEL_ORDER';

export const SAVE_CREDENTIAL = 'SAVE_CREDENTIAL';

export const ActionSuccessfully = (action: string): string => {
  return action.concat(SUCCESS_SUFFIX);
};

export const ActionFailure = (action: string): string => {
  return action.concat(FAILURE_SUFFIX);
};
