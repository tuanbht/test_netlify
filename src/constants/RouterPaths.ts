export const ORDER_PATH = '/:orderId/:token';
export const ORDER_EXPIRED_PATH = '/order-expired';
export const ORDER_CANCELLED_PATH = '/order-cancelled';
export const NOT_FOUND_PATH = '/';
