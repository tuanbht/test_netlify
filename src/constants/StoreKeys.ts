export const ORDER_ID = 'ORDER_ID';
export const PAYMENT_TOKEN = 'PAYMENT_TOKEN';
export const ORDER_DETAILS = 'ORDER_DETAILS';
