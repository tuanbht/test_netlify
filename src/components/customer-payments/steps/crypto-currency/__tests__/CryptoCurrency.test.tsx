import React from 'react';
import { shallow } from 'enzyme';
import CryptoCurrency from '../index';
import { CRYPTO_CURRENCIES } from '../../../../../constants/CustomerPayments';

describe('CryptoCurrency', () => {
  const selectedCrypto = jest.fn();
  const container = shallow(<CryptoCurrency selectedCrypto={selectedCrypto} />);
  const { BITCOIN, ETHEREUM, USDT } = CRYPTO_CURRENCIES;
  it('renders template correctly', () => {
    expect(container).toMatchSnapshot();
  });

  describe('select a crypto currency', () => {
    describe('select BITCOIN crypto', () => {
      beforeEach(() => {
        const button = container.find({ children: [container.find(BITCOIN.logo), BITCOIN.fullName] });
        button.simulate('click');
      });

      it('should not be called with crypto BITCOIN', () => {
        expect(selectedCrypto).not.toBeCalledWith(CRYPTO_CURRENCIES.BITCOIN);
      });
    });

    describe('select ETHEREUM crypto', () => {
      beforeEach(() => {
        const button = container.find({ children: [container.find(ETHEREUM.logo), ETHEREUM.fullName] });
        button.simulate('click');
      });

      it('should be called with crypto ETHEREUM', () => {
        expect(selectedCrypto).toBeCalledWith(CRYPTO_CURRENCIES.ETHEREUM);
      });
    });

    describe('select USD TETHER crypto', () => {
      beforeEach(() => {
        const button = container.find({ children: [container.find(USDT.logo), USDT.fullName] });
        button.simulate('click');
      });

      it('should not be called with crypto USD TETHER', () => {
        expect(selectedCrypto).not.toBeCalledWith(CRYPTO_CURRENCIES.USDT);
      });
    });
  });
});
