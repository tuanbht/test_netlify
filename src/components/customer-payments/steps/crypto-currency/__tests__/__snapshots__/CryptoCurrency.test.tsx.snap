// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CryptoCurrency renders template correctly 1`] = `
<div>
  <WithStyles(ForwardRef(Grid))
    container={true}
    justify="space-between"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <div
        className="makeStyles-merchantName-1"
      >
        Ohana
      </div>
      <div
        className="makeStyles-orderName-2"
      >
        Order #8867
      </div>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <span
        className="makeStyles-amount-3"
      >
        $76.25 USD
      </span>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Divider))
    className="makeStyles-divider-4"
  />
  <div
    className="makeStyles-selectCrypto-5"
  >
    <div
      className="makeStyles-selectTitle-6"
    >
      Select a Cryptocurrency
    </div>
    <WithStyles(ForwardRef(Button))
      className="makeStyles-root-8 makeStyles-root-9"
      disabled={true}
    >
      <ForwardRef(SvgIconBtc)
        className="makeStyles-iconCrypto-7"
      />
      Bitcoin
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      className="makeStyles-root-8 makeStyles-root-10"
      onClick={[Function]}
    >
      <ForwardRef(SvgIconEth)
        className="makeStyles-iconCrypto-7"
      />
      Ethereum
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      className="makeStyles-root-8 makeStyles-root-11"
      disabled={true}
    >
      <ForwardRef(SvgIconUsdt)
        className="makeStyles-iconCrypto-7"
      />
      USD Tether
    </WithStyles(ForwardRef(Button))>
  </div>
</div>
`;
